DVCodingStyle.md

```diff
@@ -461,6 +462,125 @@ utility macros for even more common code sequences.
+
+    *   If macros are defined for use in multiple sources, place them in a
+        separate SystemVerilog header file with appropriate `ifdef` guards.
+
+        This header file must not be compilation unit (provided as an input to
+        the tool for compilation). Instead, it should be `` `included`` in the
+        package, module or the interface that intends to invoke them.
+
+        Macro names should be appropriately prefixed with a string to avoid
+        potential namespace collisions. It is typical to use the file name for
+        the prefix string.
+
+        :-1:
+        ```systemverilog
+        `define STRINGIFY(s) `"s`"
+        // Gotcha! UVM library code also defines a macro with the same name.
+        // Depending on the tool, this may be flagged as an error. You may need
+        // this to support non-UVM testbench designs.
+        ```
+        :+1:
+        ```systemverilog
+        // file: dv_macros.h
+        `define DV_STRINGIFY(s) `"s`"
+        ```
+    *   If the scope of the macro usage is limited to the file, `undef` them
+        at the end of the file.
+
+        Macros have a global scope regardless of where they are defined. Some
+        tools do not even warn about namespace collisions, causing unnecessary
+        debug overhead. It is a good practice to `undef` macros at the end of
+        the file so that they are no longer visible to all subsequent
+        compilation units.
+3.  Avoid passing expressions as formal macro arguments.
+
+    As illustrated in the example below, it is better to avoid passing
+    expressions as arguments, since they are replaced verbatim. The macros may
+    reference the input arguments in multiple places, casuing the expressions to
+    be evaluated multiple times, resulting in unforeseen side effects.
+
+    :-1:
+    ```systemverilog
+    `define CHECK_AND_UPDATE(a, b) if (a != b) a = b;
+
+    `CHECK_AND_UPDATE(x, y++)
+    // Gotcha! This results in: if (x != y++) x = y++; x and y now have totally
+    // different and unexpected values.
+    ```
+    :+1:
+    ```systemverilog
+    `define CHECK_AND_UPDATE(a, b) if (a != b) a = b;
+
+    y++;
+    `CHECK_AND_UPDATE(x, y)
+    ```
+4.  Wrap multi-line macros in `begin` and `end`.
+
+    Multi-line macros are typically problematic when invoked in a conditional
+    statement. Here's an example:
+
+    :-1:
+    ```systemverilog
+    `define UPDATE_VALUES(a, b) \
+      a = 88;                   \
+      b = 1955;
+
+    if (power == 1.21) `UPDATE_VALUES(speed, year)
+    // Gotcha! The year is unconditionally set to 1955.
+    ```
+    :+1:
+    ```systemverilog
+    `define UPDATE_VALUES(a, b) \
+      begin                     \
+        a = 88;                 \
+        b = 1955;               \
+      end
+
+    if (power == 1.21) `UPDATE_VALUES(speed, year)
+    ```
+5.  Never invoke macros on the same line as the conditonal statement it is invoked
```

Comment:
Nit: conditional.

Also, is this actually necessary? If people are following the previous instruction, this makes no difference.

Reply:
Yeah, removed.